hostsfile/src/lib.rs

```diff
@@ -171,6 +171,24 @@ impl HostsBuilder {
         Ok(hosts_dir.with_file_name(temp_filename))
     }
 
+    fn has_content_changed(hosts_path: &Path, new_contents: &[u8]) -> io::Result<bool> {
+        let hosts_file = File::open(hosts_path)?;
+
+        if hosts_file.metadata()?.len() != new_contents.len() as u64 {
+            return Ok(true);
+        }
+
+        let hosts_buf = BufReader::new(hosts_file);
+
+        for (b1, b2) in hosts_buf.bytes().zip(new_contents) {
+            if b1.unwrap() != *b2 {
+                return Ok(true);
+            }
+        }
```

Comment:
Thanks for the suggestion! It also requires a `!` before it though, since the logic is inversed.

Reply:
Oops, yep you're right. I'll let you decide which method you'd prefer. The nice thing about `Iterator::eq` is it will include the length check, though I understand why you're doing the length check ahead of time so you don't have to read the file.

The length check before reading could _theoretically_ be racy, but nothing to worry about. To trigger it, I think a new `/etc/hosts` would have to be written which has a truncated version of `new_contents` in between the metadata length check and the byte comparison. The end result would be that you just don't write the file at all, which is no big deal I suppose.